sway/tree/view.c

```diff
@@ -368,14 +368,11 @@ void view_set_activated(struct sway_view *view, bool activated) {
 
 void view_request_activate(struct sway_view *view) {
 	struct sway_workspace *ws = view->container->pending.workspace;
-	if (!ws) { // hidden scratchpad container
-		return;
-	}
 	struct sway_seat *seat = input_manager_current_seat();
 
 	switch (config->focus_on_window_activation) {
 	case FOWA_SMART:
```

Comment:
You need to handle the FOWA_FOCUS case as well. Revealing the scratchpad container is probably the way most consistent with the ususal behavior of sway. So it will need at least something like:

```c
if (container_is_scratchpad_hidden_or_child(view->container)) {
    root_scratchpad_show(view->container);
}
```

Reply:
Hi there,

I have added the suggested line. Please tell me if this is correct.